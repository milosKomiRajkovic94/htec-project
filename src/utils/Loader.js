import React from "react";

import { BeatLoader } from "react-spinners";

import PropTypes from "prop-types";

const Loader = ({
  classNameOfWrapper,
  classNameOfLoader,
  classNameOfText,
  text,
}) => {
  return (
    <div
      className={
        classNameOfWrapper === undefined || classNameOfWrapper === null
          ? "loaderWrapperDefaultStyle"
          : `loaderWrapperDefaultStyle ${classNameOfWrapper} `
      }
    >
      <BeatLoader
        loading
        className={
          classNameOfLoader === undefined || classNameOfLoader === null
            ? "loaderDefaultStyle"
            : `loaderDefaultStyle ${classNameOfLoader}`
        }
      />
      {text === undefined || text === null ? (
        ""
      ) : (
        <p
          className={
            classNameOfText === undefined || classNameOfLoader === null
              ? "loaderTextDefaultStyle"
              : `loaderTextDefaultStyle ${classNameOfText}`
          }
        >
          {text}
        </p>
      )}
    </div>
  );
};

Loader.propTypes = {
  classNameOfWrapper: PropTypes.string,
  classNameOfLoader: PropTypes.string,
  classNameOfText: PropTypes.string,
  text: PropTypes.string,
};

Loader.defaultProps = {
  classNameOfWrapper: "",
  classNameOfLoader: "",
  classNameOfText: "",
  text: "",
};

export default Loader;
