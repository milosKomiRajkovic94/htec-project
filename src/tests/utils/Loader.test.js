import React from "react";

import ReactDOM from "react-dom";

import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Loader from "../../utils/Loader";

afterEach(cleanup);

it("renders without crashing without props", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Loader />, div);
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <Loader
        singleArticleData={{
          title: "something",
          content: "something",
          url: "something",
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
