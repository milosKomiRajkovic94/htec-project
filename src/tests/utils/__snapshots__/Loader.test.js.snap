// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="loaderWrapperDefaultStyle  "
>
  <div
    className="css-0"
  >
    <div
      className="css-110do72"
    />
    <div
      className="css-1by9rry"
    />
    <div
      className="css-110do72"
    />
  </div>
  <p
    className="loaderTextDefaultStyle "
  >
    
  </p>
</div>
`;
